service/controller/v11/resource/cloudformation/update.go

```diff
@@ -28,26 +30,26 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 
 		if stackStateToUpdate.ShouldUpdate && !stackStateToUpdate.ShouldScale {
 			// Fetch the etcd volume information.
-			etcdVolume := true
-			persistentVolume := false
-			volumes, err := r.ebs.ListVolumes(customObject, etcdVolume, persistentVolume)
+			filterFuncs := []func(t *ec2.Tag) bool{
+				ebs.NewDockerVolumeFilter(customObject),
+				ebs.NewEtcdVolumeFilter(customObject),
+			}
+			volumes, err := r.ebs.ListVolumes(customObject, filterFuncs...)
 			if err != nil {
 				return microerror.Mask(err)
 			}
-			if len(volumes) != 1 {
-				return microerror.Maskf(executionFailedError, "there must be 1 volume for etcd, got %d", len(volumes))
-			}
```

Comment:
I found this weird and removed it. Uncertain if we should keep it though. There are anyway transitions in which there are one or two so this was not really sufficient anyway.